.github/workflows/docker-build.yml

```diff
@@ -29,7 +29,8 @@ jobs:
         with:
           images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
           tags: |
-            type=sha
+            type=sha,format=long,enable=${{ github.ref == format('refs/heads/{0}', 'main') }}
+            type=ref,enable=${{ github.ref != format('refs/heads/{0}', 'main') }},event=branch
```

Comment:
How will the resulting tag look like?

Reply:
if the branch is main the docker tag will be sha-xxxxxxx, for any other branch the docker tag is the branch name (that way we can create a matrix of testing regardless of where the docker image is coming from and we're also able to differentiate between them)

Reply:
yes, that is correct

Reply:
I am fine with this, but don't understand the usecase behind this, can you elaborate?